### DISCUSSION TITLE
multiple installation of the same jacobalberty/unifi container

### DISCUSSION THREAD

**Discussion by avsion** [containrrr/watchtower#642] [Q&A] [answered]
Hi, every time there is a new jacobalberty/unifi version i can see multiple attempts to install of the same jacobalberty/unifi container until it succeed, see below log. i also can see it keeps puling image and creating new volume. any ideas what wrong? 

i have used this command to install the container
docker run --rm --init -p 8080:8080 -p 8443:8443 -p 3478:3478/udp -p 10001:10001/udp -e TZ='Australia/Sydney' -v ~/unifi:/unifi --name unifi jacobalberty/unifi:latest

lsb_release -a                                                                                                                                                                    130 ↵
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal

docker -v
Docker version 19.03.12, build 48a66213fe

Thank you

time="2020-07-04T06:48:14Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:a06a1ed1b0295534f218a4dc2d8eba8a24f77d48f6e9ca8ee82d122a875889c7)"
time="2020-07-04T06:48:22Z" level=info msg="Stopping /unifi (3e97d5f7f24d2a42bb5fb8d1ae39b861e255c8952919ec657f267bad5322430c) with SIGTERM"
time="2020-07-04T06:48:29Z" level=info msg="Creating /unifi"
time="2020-07-04T07:03:02Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:5c2f25bf41651394afe9c97aa6b296ed98ff9f87f9a103539c2b9bf97b9d2c5a)"
time="2020-07-04T07:03:12Z" level=info msg="Stopping /unifi (4fc74f417aa5b9ce0308583fb6e39761c18803b17009ff524630b5ea2a2e2f37) with SIGTERM"
time="2020-07-04T07:03:17Z" level=info msg="Creating /unifi"
time="2020-07-04T16:48:02Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:9d0f315030485e91baa8707f7cda0b4c49afb482ccdfeafbe27383af5aef4164)"
time="2020-07-04T16:48:10Z" level=info msg="Stopping /unifi (58345a3d22a72e12427d6ab086e7d09f290bca5e80e5b4d09b8c54be6df7277b) with SIGTERM"
time="2020-07-04T16:48:15Z" level=info msg="Creating /unifi"
time="2020-07-04T16:53:02Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:567b207048513bdc51fbb2a74569d468250917b642c304d4c6f7937b92754669)"
time="2020-07-04T16:53:10Z" level=info msg="Stopping /unifi (94a8f11b1b09c771ae81246753ec71c7e5db0b67ee3d49ea78e9118a7e1428ae) with SIGTERM"
time="2020-07-04T16:53:15Z" level=info msg="Creating /unifi"
time="2020-08-21T14:57:51Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:a3f768ab56e574a1438d5657c11aa735791e7679dbdd63c52f72568244709855)"
time="2020-08-21T14:57:55Z" level=info msg="Stopping /unifi (5f4e7a461b6a398f67b6ed7b5fcfa28a4df7f7e24e80157c571ff60a2efd8714) with SIGTERM"
time="2020-08-21T14:58:00Z" level=info msg="Creating /unifi"
time="2020-08-21T15:07:43Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:1198ae7760bf3084813c12678124cd6f1dcb1668f251d503dd0b285f33d59a66)"
time="2020-08-21T15:07:50Z" level=info msg="Stopping /unifi (0f80e7236dbffa8e49e677f261d2bd861b6db2109432287239683b8e2e5b9438) with SIGTERM"
time="2020-08-21T15:07:57Z" level=info msg="Creating /unifi"
time="2020-08-21T15:12:43Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:0b218ca78ebbd3d6fa567697ba9631ce09f1a683f38e4922a91cb7a3bc114c55)"
time="2020-08-21T15:12:51Z" level=info msg="Stopping /unifi (26491661458f4cecfe06465f3dd93c500ab96319dc41b4e21a457e88aae49bde) with SIGTERM"
time="2020-08-21T15:12:56Z" level=info msg="Creating /unifi"
time="2020-08-25T14:57:42Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:a60c09d3543ac2ab334f74da219de823cc4aaf851f13b5001cea727965dbe841)"
time="2020-08-25T14:57:50Z" level=info msg="Stopping /unifi (d42f7dcd2099b4c34e9545b33ec07dd63cefe03ae54087f9090546d743cb58be) with SIGTERM"
time="2020-08-25T14:57:55Z" level=info msg="Creating /unifi"
time="2020-08-25T15:07:38Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:730b799f18b411cfb227277ecac0c2db37063d713172040c203f74baad11e8ce)"
time="2020-08-25T15:07:46Z" level=info msg="Stopping /unifi (b87c02cfaa7ff892fdae10c227de3e58cfbcc4a251475bae6f5250e80cb70a62) with SIGTERM"
time="2020-08-25T15:07:51Z" level=info msg="Creating /unifi"
time="2020-09-11T16:42:39Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:aff106451b2f14f15462211f2568def9016042c0e48b395ef141e2b6d6d075c7)"
time="2020-09-11T16:42:47Z" level=info msg="Stopping /unifi (45acc9f24f13f95d8b2bc2cee315e22a865fb48dbbaa5903844b4757029f8c34) with SIGTERM"
time="2020-09-11T16:42:51Z" level=info msg="Creating /unifi"
time="2020-09-11T16:52:38Z" level=info msg="Found new jacobalberty/unifi:latest image (sha256:4a8af2958c92ad3b66378b76f2c39dd71603e1faf7466b40ca4640a42da2028f)"
time="2020-09-11T16:52:45Z" level=info msg="Stopping /unifi (f50909b76a4c4db6cf3a333adc471592917f6b6b359da41fafa8c39a0a889673) with SIGTERM"
time="2020-09-11T16:52:50Z" level=info msg="Creating /unifi"

also getting this errors from docker, not sure if its related,

Sep 10 18:27:04 docker dockerd[905]: time="2020-09-10T18:27:04.885111581Z" level=info msg="Attempting next endpoint for pull after error: Get https://registry-1.docker.io/v2/containrrr/wa>
Sep 10 18:27:04 docker dockerd[905]: time="2020-09-10T18:27:04.885361517Z" level=error msg="Handler for POST /v1.25/images/create returned error: Get https://registry-1.docker.io/v2/conta>
Sep 11 16:42:51 docker dockerd[905]: time="2020-09-11T16:42:51.124640502Z" level=info msg="ignoring event" module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Sep 11 16:52:50 docker dockerd[905]: time="2020-09-11T16:52:50.171276409Z" level=info msg="ignoring event" module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Sep 11 18:07:04 docker dockerd[905]: time="2020-09-11T18:07:04.496352551Z" level=info msg="Attempting next endpoint for pull after error: Get https://registry-1.docker.io/v2/containrrr/wa>
Sep 11 18:07:04 docker dockerd[905]: time="2020-09-11T18:07:04.496648936Z" level=error msg="Handler for POST /v1.25/images/create returned error: Get https://registry-1.docker.io/v2/conta>
Sep 12 06:07:05 docker dockerd[905]: time="2020-09-12T06:07:05.479393411Z" level=info msg="Attempting next endpoint for pull after error: Get https://registry-1.docker.io/v2/containrrr/wa>
Sep 12 06:07:05 docker dockerd[905]: time="2020-09-12T06:07:05.480693594Z" level=error msg="Handler for POST /v1.25/images/create returned error: Get https://registry-1.docker.io/v2/conta>
Sep 12 11:32:12 docker dockerd[905]: time="2020-09-12T11:32:12.665837439Z" level=info msg="Attempting next endpoint for pull after error: Get https://registry-1.docker.io/v2/portainer/por>
Sep 12 11:32:12 docker dockerd[905]: time="2020-09-12T11:32:12.666013926Z" level=error msg="Handler for POST /v1.25/images/create returned error: Get https://registry-1.docker.io/v2/porta>
~

**Comment by zoispag** [comment, marked as answer]
Well, these look like real images to me:

```
2020-07-04 06:48:14: Found new jacobalberty/unifi:latest image (sha256:a06a1ed1b0295534f218a4dc2d8eba8a24f77d48f6e9ca8ee82d122a875889c7)
2020-07-04 07:03:02: Found new jacobalberty/unifi:latest image (sha256:5c2f25bf41651394afe9c97aa6b296ed98ff9f87f9a103539c2b9bf97b9d2c5a)
2020-07-04 16:48:02: Found new jacobalberty/unifi:latest image (sha256:9d0f315030485e91baa8707f7cda0b4c49afb482ccdfeafbe27383af5aef4164)
2020-07-04 16:53:02: Found new jacobalberty/unifi:latest image (sha256:567b207048513bdc51fbb2a74569d468250917b642c304d4c6f7937b92754669)
2020-08-21 14:57:51: Found new jacobalberty/unifi:latest image (sha256:a3f768ab56e574a1438d5657c11aa735791e7679dbdd63c52f72568244709855)
2020-08-21 15:07:43: Found new jacobalberty/unifi:latest image (sha256:1198ae7760bf3084813c12678124cd6f1dcb1668f251d503dd0b285f33d59a66)
2020-08-21 15:12:43: Found new jacobalberty/unifi:latest image (sha256:0b218ca78ebbd3d6fa567697ba9631ce09f1a683f38e4922a91cb7a3bc114c55)
2020-08-25 14:57:42: Found new jacobalberty/unifi:latest image (sha256:a60c09d3543ac2ab334f74da219de823cc4aaf851f13b5001cea727965dbe841)
2020-08-25 15:07:38: Found new jacobalberty/unifi:latest image (sha256:730b799f18b411cfb227277ecac0c2db37063d713172040c203f74baad11e8ce)
2020-09-11 16:42:39: Found new jacobalberty/unifi:latest image (sha256:aff106451b2f14f15462211f2568def9016042c0e48b395ef141e2b6d6d075c7)
2020-09-11 16:52:38: Found new jacobalberty/unifi:latest image (sha256:4a8af2958c92ad3b66378b76f2c39dd71603e1faf7466b40ca4640a42da2028f)
```

Every SHA is different, which makes me thing that something weird is happening with their build process.

I see for example that `latest` (and all tags) are built twice from the same commit hash:
![image](https://user-images.githubusercontent.com/21138205/93019470-7fda2d00-f5d7-11ea-8727-ac50790a1c44.png)

This probably explains what you are experiencing.

**Reply by avsion** [reply]
great insight, will check with jacobalberty

Thank you
